pkg/services/serviceaccounts/api/api.go

```diff
@@ -40,9 +43,27 @@ func (api *ServiceAccountsAPI) RegisterAPIEndpoints(
 	auth := acmiddleware.Middleware(api.accesscontrol)
 	api.RouterRegister.Group("/api/serviceaccounts", func(serviceAccountsRoute routing.RouteRegister) {
 		serviceAccountsRoute.Delete("/:serviceAccountId", auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(serviceaccounts.ActionDelete, serviceaccounts.ScopeID)), routing.Wrap(api.DeleteServiceAccount))
+		serviceAccountsRoute.Post("/", auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(serviceaccounts.ActionCreate, serviceaccounts.ScopeID)), routing.Wrap(api.CreateServiceAccount))
 	})
 }
 
+// POST /api/serviceaccounts
+func (api *ServiceAccountsAPI) CreateServiceAccount(c *models.ReqContext) response.Response {
+	cmd := serviceaccounts.CreateServiceaccountForm{}
+	if err := web.Bind(c.Req, &cmd); err != nil {
+		return response.Error(http.StatusBadRequest, "Bad request data", err)
+	}
+	user, err := api.service.CreateServiceAccount(c.Req.Context(), &cmd)
+	switch {
+	case errors.Is(err, serviceaccounts.ErrServiceAccountNotFound):
+		return response.Error(http.StatusBadRequest, "Failed to create role with the provided name", err)
+	case err != nil:
+		return response.Error(http.StatusInternalServerError, "Failed to create service account", err)
+	}
+
+	return response.JSON(http.StatusCreated, user).SetHeader("Location", fmt.Sprintf("/api/serviceaccounts/%d", user.Id))
```

Comment:
@jeremy-watkins are you referring to this?

I am not sure about your comment around making a new name for it. As I get 201 Created.
Maybe you are referring to something else here


we are also doing this in the access control package in enterprise

Reply:
wrong Jeremy :D

Yes, I thought that the comment would be attached to the code.  Odd.

Not the returned name, the name in the code.  I don't see the benefit of writing "http.StatusCreated" when you could just write 201.

It's shorter, it's well know, and it's a constant.

We're not in the accesscontrol package, we don't have to follow what they do.